sklearn/ensemble/_stacking.py

```diff
@@ -406,6 +406,10 @@ class StackingClassifier(ClassifierMixin, _BaseStacking):
         list is defined as a tuple of string (i.e. name) and an estimator
         instance. An estimator can be set to 'drop' using `set_params`.
 
+        The type of estimator is generally expected to be a classifier.
+        However, one can pass a regressor for some use case (e.g. ordinal
+        regression) but the target `y` is expected to be encoded as numerical.
+
```

Comment:
I think that we can add a Notes section below (just before the Examples section).

```
Notes
-----
In some cases (e.g. ordinal regression), one can pass regressors as the first layer
of the :class:`StackingClassifier`. However, note that `y` will be internally
encoded in a numerically increasing order or lexicographic order. If this ordering
is not adequate, one should manually numerically encode the classes in the desired
order.
```

Reply:
Ok, done. It already existed before the References section

Reply:
@glemaitre can we merge?